service/history/queuev2/virtual_queue.go

```diff
@@ -46,6 +46,9 @@ type (
 		GetState() []VirtualSliceState
 		UpdateAndGetState() []VirtualSliceState
 		MergeSlices(...VirtualSlice)
+		IterateSlices(func(VirtualSlice))
+		ClearSlices(func(VirtualSlice) bool)
+		SplitSlices(func(VirtualSlice) (remaining []VirtualSlice, split bool))
```

Comment:
Better to name these arguments (to give a hint of what they are) and some inline comments to explain these funcs. Actually same would be great for other non-obvious methods in this interface